src/metabase/util/date_2.clj

```diff
@@ -431,21 +431,16 @@
                      :exclusive (add t resolution -1)))}
      :=  (range t unit options))))
 
-;; Moving the type hints to the arg lists makes clj-kondo happy, but breaks eastwood (and maybe causes reflection
-;; warnings) at the call sites.
-#_{:clj-kondo/ignore [:non-arg-vec-return-type-hint]}
-(defn ^PeriodDuration period-duration
+(defn period-duration
```

Comment:
👍 

I missed this earlier, I'm guessing this is where you the linter changes came from.

I think the reason Eastwood was fussy here was because we had custom `:arglists` metadata with duplicate arities which is a little questionable even if it is done for documentation purposes. 

Your change seems like a good way to fix it altho it was nice to know what `x` and `y` were allowed to mean, maybe you can add that to the docstring so we don't lose that documentation.

Reply:
Actually after this change Eastwood was still complaining, so I've removed the change and left it as it was before.